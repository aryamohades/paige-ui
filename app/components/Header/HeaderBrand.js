import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const StyledLink = styled(Link)`
  text-decoration: none;
  color: #222;
  font-size: 21px;
  font-weight: bold;
  padding: 1em;
  margin-right: 0.5em;
`;

const HeaderBrand = ({ text }) => <StyledLink to="/">{text}</StyledLink>;

HeaderBrand.propTypes = {
  text: PropTypes.string.isRequired,
};

export default HeaderBrand;
