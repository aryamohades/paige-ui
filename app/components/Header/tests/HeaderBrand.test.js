import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import HeaderBrand from '../HeaderBrand';

describe('<HeaderBrand />', () => {
  it('should match the snapshot', () => {
    const text = 'test';
    const renderedComponent = renderer
      .create(
        <Router>
          <HeaderBrand text={text} />
        </Router>,
      )
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render the correct text', () => {
    const text = 'test';
    const renderedComponent = mount(
      <Router>
        <HeaderBrand text={text} />
      </Router>,
    );
    expect(renderedComponent.find(HeaderBrand).text()).toEqual(text);
  });
});
