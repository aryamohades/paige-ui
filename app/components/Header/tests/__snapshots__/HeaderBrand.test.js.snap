// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBrand /> should match the snapshot 1`] = `
<a
  className="HeaderBrand__StyledLink-Bfuoh IfHBM"
  href="/"
  onClick={[Function]}
>
  test
</a>
`;
